### PR TITLE
detect-it-easy: Add version 3.03

### PR DESCRIPTION
**Pull request by YenForYang**
https://github.com/horsicq/Detect-It-Easy

### CODE REVIEW

File: bucket/detect-it-easy.json [hunk 1]
```diff
@@ -0,0 +1,34 @@
+{
+    "version": "3.01",
+    "description": "Program for determining types of files.",
+    "homepage": "http://ntinfo.biz",
+    "license": "MIT",
+    "architecture": {
+        "64bit": {
+            "url": "https://github.com/horsicq/DIE-engine/releases/download/3.01/die_win64_portable_3.01.zip",
+            "hash": "65B7731D23C34CEE5D2EDBDC0D02CAF2D9363BE41CE1D1C2B0D23534B99EBD10"
```
Comment (earnestma) [inline bucket/detect-it-easy.json:9]:
```suggestion
            "hash": "65b7731d23c34cee5d2edbdc0d02caf2d9363be41ce1d1c2b0d23534b99ebd10"
```

File: bucket/detect-it-easy.json [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+        },
+        "32bit": {
+            "url": "https://github.com/horsicq/DIE-engine/releases/download/3.01/die_win32_portable_3.01.zip",
+            "hash": "4274DC088543216F6305414BC1B46D56EA51E6A952CB657684753755541DC8B6"
```
Comment (earnestma) [inline bucket/detect-it-easy.json:13]:
```suggestion
            "hash": "4274dc088543216f6305414bc1b46d56ea51e6a952cb657684753755541dc8b6"
```

File: bucket/detect-it-easy.json [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+        }
+    },
+    "bin": [ "die.exe", "diec.exe" ],
+    "shortcuts": [
+        ["die.exe", "Detect It Easy"]
+    ],
```
Comment (earnestma) [inline bucket/detect-it-easy.json:19]:
```suggestion
    "shortcuts": [
        [
            "die.exe",
            "Detect It Easy"
        ]
    ],
```

File: bucket/detect-it-easy.json [hunk 4]
```diff
@@ -0,0 +1,34 @@
+{
+    "version": "3.01",
+    "description": "Program for determining types of files.",
+    "homepage": "http://ntinfo.biz",
+    "license": "MIT",
+    "architecture": {
+        "64bit": {
+            "url": "https://github.com/horsicq/DIE-engine/releases/download/3.01/die_win64_portable_3.01.zip",
+            "hash": "65B7731D23C34CEE5D2EDBDC0D02CAF2D9363BE41CE1D1C2B0D23534B99EBD10"
+        },
```
Comment (earnestma) [inline bucket/detect-it-easy.json:10]:
```suggestion
            "extract_dir": "die_win64_portable"
        },
```

File: bucket/detect-it-easy.json [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+        "32bit": {
+            "url": "https://github.com/horsicq/DIE-engine/releases/download/3.01/die_win32_portable_3.01.zip",
+            "hash": "4274DC088543216F6305414BC1B46D56EA51E6A952CB657684753755541DC8B6"
+        }
```
Comment (earnestma) [inline bucket/detect-it-easy.json:14]:
```suggestion
            "extract_dir": "die_win32_portable"
        }
```

File: bucket/detect-it-easy.json [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+    ],
+    "persist": "die.ini",
+    "checkver": {
+        "github": "https://github.com/horsicq/DIE-engine"
+    },
+    "autoupdate": {
+        "architecture": {
+            "64bit": {
+                "url": "https://github.com/horsicq/DIE-engine/releases/download/$version/die_win64_portable-$version.zip"
+            },
```
Comment (earnestma) [inline bucket/detect-it-easy.json:28]:
```suggestion
                "extract_dir": "die_win64_portable"
            },
```

File: bucket/detect-it-easy.json [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+            "32bit": {
+                "url": "https://github.com/horsicq/DIE-engine/releases/download/$version/die_win32_portable-$version.zip"
+            }
```
Comment (earnestma) [inline bucket/detect-it-easy.json:31]:
```suggestion
                "extract_dir": "die_win32_portable"
            }
```

File: bucket/detect-it-easy.json [hunk 8]
```diff
@@ -0,0 +1,34 @@
+{
+    "version": "3.01",
+    "description": "Program for determining types of files.",
```
Comment (earnestma) [inline bucket/detect-it-easy.json:3]:
```suggestion
    "description": "Program for determining types of files",
```

**Review by earnestma — CHANGES_REQUESTED** [review]
Please format/ edit your PR title as: **`detect-it-easy: Add version 3.01`**

File: bucket/detect-it-easy.json [hunk 9]
```diff
@@ -0,0 +1,34 @@
...
+            "url": "https://github.com/horsicq/DIE-engine/releases/download/3.01/die_win32_portable_3.01.zip",
+            "hash": "4274DC088543216F6305414BC1B46D56EA51E6A952CB657684753755541DC8B6"
+        }
+    },
+    "bin": [ "die.exe", "diec.exe" ],
+    "shortcuts": [
+        ["die.exe", "Detect It Easy"]
+    ],
+    "persist": "die.ini",
+    "checkver": {
```
Comment (earnestma) [inline bucket/detect-it-easy.json:21]: Cannot determine autoupdates through GitHub, repo does not use Releases

https://github.com/horsicq/Detect-It-Easy/releases

Comment (earnestma) [inline bucket/detect-it-easy.json:21]: Project does not use GitHub releases (https://github.com/horsicq/Detect-It-Easy/releases)

**Review by Ash258 — CHANGES_REQUESTED** [review]
Persisting ini file is not enough most likely

https://github.com/lukesampson/scoop-extras/pull/4295#issuecomment-645802110

File: bucket/detect-it-easy.json [hunk 10]
```diff
@@ -0,0 +1,40 @@
...
+            "hash": "f793c8ea4578cafca7543fe407f03ac50ac15850e99983f1cc5ee19b0dda7c78"
+        },
+        "32bit": {
+            "url": "https://github.com/horsicq/DIE-engine/releases/download/3.03/die_win32_portable_3.03.zip",
+            "hash": "0ce1787066ae2bad2d160aa76a2bb0f188b978b099643b15c8b0c7b58594cc1b"
+        }
+    },
+    "bin": [
+        "die.exe",
+        "diec.exe"
```
Comment (rashil2000) [inline bucket/detect-it-easy.json:18]: What is the difference between these two binaries? Do both support arguments?

**Comment by rashil2000** [comment]
Also see persistence related notes in https://github.com/ScoopInstaller/Extras/pull/4295/files

File: bucket/detect-it-easy.json (hunk 10 above)
Comment (issaclin32) [reply bucket/detect-it-easy.json]: I think only `diec.exe` accepts arguments.

**Review by rashil2000 — APPROVED** [review]

**Comment by issaclin32** [comment]
Some of the config folders mentioned in #4295 are not used anymore.
dt

File: bucket/detect-it-easy.json (hunk 10 above)
Comment (rashil2000) [reply bucket/detect-it-easy.json:23]: I think you accidentally added `die.exe` back to bin

**Review by rashil2000 — APPROVED** [review]

File: bucket/detect-it-easy.json (hunk 10 above)
Comment (issaclin32) [reply bucket/detect-it-easy.json:23]: Oops.
fixed by https://github.com/ScoopInstaller/Extras/commit/93ac9084f093363f64db8a997afc35e23b98d679

**Pull request merged**